### PR TITLE
feat: Added kafka service

### LINKED ISSUE #1563: Apache Kafka service
Hello, I think it would be nice to have a service for apache kafka.
I wanted to try to set the service up myself, but I am not sure how to add default configuration or how it could be changed by the user.

### PR DESCRIPTION
**Pull request by Alexnortung**
Here is a rough implementation of a kafka service for devenv.

I think there are definetely place for configuration options. but I am not sure how we can implement this properly. So feedback would be appreciated :)

closes #1563

### CODE REVIEW

**Comment by Alexnortung** [comment]
Okay, I figured it out, you could just pass the connectors to the start command

**Comment by sandydoo** [comment]
Take a look at https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/misc/apache-kafka.nix. You should be able to reuse large parts of this, particularly the config/settings handling, and ideally keep a similar interface.

**Comment by Alexnortung** [comment]
Hey @sandydoo thanks for the feedback!
I have now used many of the options provided in the NixOS module. However I have tweaked some defaults, so that it just works without having the user configure anything but `enable = true`.

**Comment by sandydoo** [comment]
> However I have tweaked some defaults, so that it just works without having the user configure anything but enable = true.

That's the right approach.

@Alexnortung, is there a basic test we could add for this to make sure it launches? I'm not familiar with Kafka, but maybe a ping/healthcheck/queue-unqueue type of thing. See the `examples` and `tests` directories.

**Comment by Alexnortung** [comment]
Hi @sandydoo I have Added settings to kafka connect and added a test for kafka.
I also wanted to add readiness probe to kafka, but it seems like something weird is going on when you add readiness probe to the service. Do you have any suggestions?

File: tests/kafka/.test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set -e
+
+kafka-topics.sh --list --bootstrap-server localhost:9092
```
Comment (domenkozar) [inline tests/kafka/.test.sh:3]: Where does `kafka-topics.sh` come from?

Comment (Alexnortung) [reply tests/kafka/.test.sh:3]: If you prefer we could probably also do a `curl` since the script is probably just calling an endpoint in kafka

**Comment by Alexnortung** [comment]
Am I missing something since the github actions are failing?

File: tests/kafka/.test.sh (hunk 1 above)
Comment (domenkozar) [reply tests/kafka/.test.sh:3]: yeah let's do just curl

**Comment by domenkozar** [comment]
> Am I missing something since the github actions are failing?

some of them fail, but you'll see kafka being run in the test action

**Comment by Alexnortung** [comment]
I have now also added readiness probes for kafka and kafka connect. I have also added kafka as a dependency to kafka connect, so that kafka connect doesn't fail to start if kafka is not started correctly or is using more time than usual.

**Pull request merged**